src/pages/SharePage.js

```diff
@@ -0,0 +1,175 @@
+     */
+    onCloseButtonClick() {
+        clearSharedItem();
+        redirect(this.props.currentlyViewedReportID !== ''
+            ? ROUTES.getReportRoute(this.props.currentlyViewedReportID)
+            : ROUTES.HOME);
+    }
+
+    /**
+     * Hides navigation menu on redirect to report page.
+     */
+    onLinkClick() {
+        hideSidebar();
+    }
+
+    /**
+     * Method called when a Create Menu item is selected.
+     */
+    onCreateMenuItemSelected() {
+        this.toggleCreateMenu();
+        ChatSwitcher.show();
+    }
+
+    /**
+     * Posts shared item to selected report
+     *
+     * @param {Number} reportID report id
+     */
+    addSharedItemToReport(reportID) {
+        if (!this.props.sharedItem) {
+            return;
+        }
+
+        switch (this.props.sharedItem.type) {
+            case ShareManager.TYPE.TEXT:
+                addAction(reportID, this.props.sharedItem.data);
+                break;
+            case ShareManager.TYPE.FILE:
+                addAction(reportID, '', this.props.sharedItem.data);
+                break;
+            default:
+                break;
+        }
+
+        clearSharedItem();
+    }
+
+    /**
+     * Method called when we click the floating action button
+     *
+     * Method called either when:
+     * Pressing the floating action button to open the CreateMenu modal
+     * Selecting an item on CreateMenu or closing it by clicking outside of the modal component
+     */
+    toggleCreateMenu() {
+        this.setState(state => ({
+            isCreateMenuActive: !state.isCreateMenuActive,
+        }));
+    }
+
+    render() {
+        return (
+            <SafeAreaProvider>
+                <CustomStatusBar />
+                <SafeAreaInsetsContext.Consumer style={[styles.flex1]}>
+                    {insets => (
+                        <>
+                            <View style={[styles.flex1, styles.sidebar]}>
+                                <SidebarLinks
+                                    title="Send to..."
+                                    showCloseButton
+                                    insets={insets}
+                                    onCloseButtonClick={this.onCloseButtonClick}
+                                    onLinkClick={this.onLinkClick}
+                                    onReportSelected={this.addSharedItemToReport}
+                                />
+                                <FAB
+                                    isActive={this.state.isCreateMenuActive}
+                                    onPress={this.toggleCreateMenu}
+                                />
+                            </View>
+                            <CreateMenu
+                                onClose={this.toggleCreateMenu}
+                                isVisible={this.state.isCreateMenuActive}
+                                onItemSelected={this.onCreateMenuItemSelected}
+                            />
```

Comment:
> We probably don't need the `CreateMenu` here?
> Same with the `FAB`?

@marcaaron 
`CreateMenu` and `FAB` would allow user to share not only to the chat from the list, but also to create a new chat and share there.
Of course a new chat can be created also with a search functionality of `ChatSwitcherView`, but that's probably less intuitive for the user.

Reply:
OK, I see that new Search view has been merged to the master, so most of this doesn't make sense anymore and should be reworked anyway :)